README.md
# deep-helper README
# ChatGPT Extension

This is a VS Code extension that leverages the power of ChatGPT API to provide interactive and intelligent responses to user queries.

## Features

- Seamlessly integrate ChatGPT into your VS Code environment.
- Get real-time responses from the ChatGPT API for the queries entered by the user.
- Customizable settings to configure API key, response format, and other options.

## Installation

1. Open Visual Studio Code.
2. Go to the Extensions view (`Ctrl+Shift+X` or `Cmd+Shift+X`).
3. Search for "Deep Helper" and click Install.
4. Reload the VS Code window to activate the extension.

## Usage

1. Click `Ctrl+Shift+P` search for Ai Search.
2. Enter your query.
3. Invoke the ChatGPT response by using the provided command or shortcut.
4. The extension will make an API call to the ChatGPT service and display the response in a separate panel or as a notification.


## Prerequisites

- You need to obtain an API key from the ChatGPT service. Visit their website and sign up to get an API key.
- Make sure you have an active internet connection for making API calls.

## Known Issues

- Issue 1: Lack Of Error Handling
- Issue 2: Header file are getting remove in c++ code due to pre tag

## Contributions

Contributions are welcome! If you have any suggestions, bug reports, or feature requests, please create an issue in the GitHub repository.

logo:<a href="https://www.flaticon.com/free-icons/coder" title="coder icons">Coder icons created by Sophia_Devi - Flaticon</a>


CHANGELOG.md
# Change Log

All notable changes to the "deep-helper" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

extension.js
const vscode = require("vscode");
const apiUrl = "https://api.openai.com/v1/chat/completions";
let apiKey = "";
// import fetch from 'node-fetch';
const fetch = require("node-fetch");

let config = vscode.workspace.getConfiguration();
let model = config.get("aisearch.API");
model = model.slice(1, model.length - 1);

apiKey = model;
// console.log(model);
// apiKey = model

async function getAnswer(query) {
  let headers = {
    "Content-Type": "application/json",
    Authorization: `Bearer ${apiKey}`,
  };
  let data = JSON.stringify({
    model: "gpt-3.5-turbo",
    messages: [
      { role: "system", content: "You are a helpful assistant." },
      { role: "user", content: query },
    ],
    temperature: 0.7,
  });

  const response = await fetch(apiUrl, {
    method: "POST",
    headers,
    body: data,
  });

  const responseData = await response.json();
  return responseData.choices[0].message.content;
}

async function getAnswerToQuery(query) {
  try {
    const response = await getAnswer(query);
    // console.log(response);
    return response;
  } catch (err) {
    throw new Error(err.message);
  }
}

// Usage example:

// This method is called when your extension is activated
// Your extension is activated the very first time the command is executed

/**
 * @param {vscode.ExtensionContext} context
 */
async function activate(context) {
  // console.log('Congratulations, your extension "deep-helper" is now active!');

  let disposable = vscode.commands.registerCommand(
    "deep-helper.AiSearch",
    async function () {
      let configI = vscode.workspace.getConfiguration();
      let modelI = configI.get("aisearch.API");
      apiKey = modelI;
      // Take input from the user Then we need to call chat gpt for that query
      const input = await vscode.window.showInputBox();
      vscode.window.showInformationMessage(input);
      vscode.window.showInformationMessage("Hello from Deep-Helper!");

      const panel = vscode.window.createWebviewPanel(
        "output", // Identifies the type of the webview. Used internally
        "AI Answer", // Title of the panel displayed to the user
        vscode.ViewColumn.One, // Editor column to show the new webview panel in.
        {} // Webview options. More on these later.
      );
      if (input == "" || input == null) {
        panel.webview.html = getWebviewContent("Enter Some Response ¯_(ツ)_/¯");
        return;
      }
      panel.webview.html = getWebviewContent("Wating for response (～￣▽￣)～");
      setTimeout(() => {
        let prom = getAnswerToQuery(input);
        prom.then((response) => {
          panel.webview.html = getWebviewContent(response);
        });
      }, 10000);

      //   prom.then((answer) => {
      //     console.log(answer);
      //     panel.webview.html = getWebviewContent(answer);
      //  })
      //  .catch((error) => console.error(error));
      // getAnswerToQuery("Hello world!")
      // .then((answer)=>{
      // })
      // .catch(error => console.error(error));
    }
  );

  context.subscriptions.push(disposable);
}

function getWebviewContent(input) {
  return `<!DOCTYPE html>
  <html>
  <head>
    <title>Code Example</title>
    <style>
      div {
        background-color: black;
        padding: 10px;
        width: 100%;
        color: green
      }
    </style>
  </head>
  <body>
    <h1>Code Example</h1>
    <div style="white-space: pre-wrap;">${input}</div>
  
    <script src="script.js"></script>
  </body>
  </html>`;
}
// This method is called when your extension is deactivated
function deactivate() {}

module.exports = {
  activate,
  deactivate,
};


chatgpt/app.js
const apiUrl = 'https://api.openai.com/v1/chat/completions';
const apiKey = "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";

const headers = {
  'Content-Type': 'application/json',
  'Authorization': `Bearer ${apiKey}`
};

const data = JSON.stringify({
  model: 'gpt-3.5-turbo',
  messages: [
    { role: 'system', content: 'You are a helpful assistant.' },
    { role: 'user', content: 'Hello!' },
    { role: 'user', content: 'write simple program in c++' }
],
  temperature: 0.7
});

fetch(apiUrl, {
  method: 'POST',
  headers,
  body: data
})
  .then(response => response.json())
  .then(data => {
    // Handle the API response data here
    console.log(data.choices);
  })
  .catch(error => {
    // Handle any errors that occur during the request
    console.error(error);
  });


chat-gpt.js
const vscode = require("vscode");


let config = vscode.workspace.getConfiguration()
const model = config.get("aisearch.API") 
console.log(model)
// const apiUrl = "https://api.openai.com/v1/chat/completions";
// const apiKey = "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";
// const https = require("https");

// const headers = {
//   "Content-Type": "application/json",
//   Authorization: `Bearer ${apiKey}`,
// };

// async function getAnswer(query) {
//   const data = JSON.stringify({
//     model: "gpt-3.5-turbo",
//     messages: [
//       { role: "system", content: "You are a helpful assistant." },
//       { role: "user", content: query },
//     ],
//     temperature: 0.7,
//   });
//   let options = {
//     method: "POST",
//     headers,
//     body: data,
//   };
//   const req = https.request(apiUrl, options, (res) => {
//     let data = "";
//     res.on("data", (chunk) => {
//       data += chunk;
//     });
//     res.on("end", () => {
//       const responseData = JSON.parse(data);
//       // Handle the response data here
//       console.log(responseData);
//     });
//   });

//   req.on("error", (error) => {
//     // Handle any errors that occur during the request
//     console.error(error);
//   });

//   req.end();

//   // const response = await fetch(apiUrl, {
//   //   method: "POST",
//   //   headers,
//   //   body: data,
//   // });

//   // const responseData = await response.json();
//   // return responseData.choices[0].message.content;
// }
// getAnswer("write the c++ program")































// // const apiUrl = "https://api.openai.com/v1/chat/completions";
// // const apiKey = "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";

// // const headers = {
// //   "Content-Type": "application/json",
// //   Authorization: `Bearer ${apiKey}`,
// // };

// // async function getAnswer(query) {
// //   const data = JSON.stringify({
// //     model: "gpt-3.5-turbo",
// //     messages: [
// //       { role: "system", content: "You are a helpful assistant." },
// //       { role: "user", content: query },
// //     ],
// //     temperature: 0.7,
// //   });

// //   const response = await fetch(apiUrl, {
// //     method: "POST",
// //     headers,
// //     body: data,
// //   });

// //   const responseData = await response.json();
// //   return responseData.choices[0].message.content;
// // }

// // async function getAnswerToQuery(query) {
// //   const response = await getAnswer(query);
// //   return response;
// // }

// // // Usage example:

// // function getCall(query) {
// //     getAnswerToQuery(query)
// //    .then((answer) => {
// //      console.log(answer);
// //   })
// //   .catch((error) => console.error(error));

// // }

// // // const headers = {
// // //   "Content-Type": "application/json",
// // //   Authorization: `Bearer ${apiKey}`,
// // // };
// // // async function getAnswer(query) {
// // //     const data = JSON.stringify({
// // //       model: "gpt-3.5-turbo",
// // //       messages: [
// // //         { role: "system", content: "You are a helpful assistant." },
// // //         { role: "user", content: query },
// // //       ],
// // //       temperature: 0.7,
// // //     });
// // //     let response =await fetch(apiUrl, {
// // //       method: "POST",
// // //       headers,
// // //       body: data,
// // //     });

// // //     return response.json();

// // //   //     .then((response) => response.json())
// // //   //     .then(async (data) => {
// // //   //       // Handle the API response data here
// // //   //       const response = await data.choices[0].message.content;
// // //   //       console.log(response);
// // //   //       return response.json;
// // //   //     })
// // //   //     .catch((error) => {
// // //   //       // Handle any errors that occur during the request
// // //   //       console.error(error);
// // //   //       return new error("Something went Wrong");
// // //   //     });

// // //   //   return { success: false };
// // // }

// // // const mainfunc = async function(){
// // //     const response = await getAnswer("hello world");
// // //     console.log(response);
// // //     console.log(response.choices[0].message.content);
// // //     return (response.choices[0].message.content);
// // // }
// // // console.log(mainfunc());
